pkg/codegen/nodejs/gen.go

```diff
@@ -393,7 +419,14 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 			stateParam, stateRef = fmt.Sprintf("state?: %s, ", stateType), "<any>state, "
 		}
 
-		fmt.Fprintf(w, "    public static get(name: string, id: pulumi.Input<pulumi.ID>, %sopts?: pulumi.CustomResourceOptions): %s {\n", stateParam, name)
+		var optionsType string
+		switch {
+		case r.IsComponent:
+			optionsType = "pulumi.ComponentResourceOptions"
+		default:
+			optionsType = "pulumi.CustomResourceOptions"
+		}
+		fmt.Fprintf(w, "    public static get(name: string, id: pulumi.Input<pulumi.ID>, %sopts?: %s): %s {\n", stateParam, optionsType, name)
```

Comment:
Note that `ComponentResource`s shouldn't have `get` functions. We can address this in a followup change.

Reply:
I'll go ahead and fix it here

Reply:
We'll need to fix this for Python as well.

Reply:
Fixed in https://github.com/pulumi/pulumi/pull/5459